app/models/govuk_publishing_components/component_doc.rb

```diff
@@ -0,0 +1,34 @@
+module GovukPublishingComponents
+  ComponentDoc = Struct.new(:id, :name, :description, :body, :fixtures) do
```

Comment:
This logic has largely been ported from https://github.com/alphagov/govuk-component-guide/

https://github.com/alphagov/govuk-component-guide/blob/master/app/models/component.rb